GitUI/Editor/Diff/AnsiEscapeUtilities.cs

```diff
@@ -0,0 +1,487 @@
+                case 2: // dim
+                case 3: // italic
+                    dim = true;
+                    foreColor = Get8bitColor(currentColorId + GetBoldDimOffset(), out _);
+                    break;
+                case 7: // reverse
+                    reverse = true;
+                    break;
+                case 39: // Default foreground color
+                    foreColor = null;
+                    currentColorId = 0;
+                    break;
+                case 49: // Default background color
+                    backColor = null;
+                    break;
+                case >= 30 and <= 37: // Set foreground color
+                case >= 90 and <= 97: // Set bright foreground color
+                    bold = bold || escapeCodes[i] >= 90;
+                    currentColorId = escapeCodes[i] - (escapeCodes[i] >= 90 ? 90 : 30);
+                    foreColor = Get8bitColor(currentColorId + GetBoldDimOffset(), out _);
+                    break;
+                case >= 40 and <= 47: // Set background color
+                case >= 100 and <= 107: // Set bright background color
+                    bold = bold || escapeCodes[i] >= 100;
+                    int backColorId = escapeCodes[i] - (escapeCodes[i] >= 100 ? 100 : 40);
+                    backColor = Get8bitColor(backColorId + GetBoldDimOffset(), out _);
+                    break;
+                case 38: // Set foreground color with sequence
+                case 48: // Set background color with sequence
+                    if (i >= escapeCodes.Count - 2)
+                    {
+                        Trace.WriteLine($"Unexpected too few arguments for {i} {escapeCodes}");
+                        DebugHelpers.Fail($"Unexpected too few arguments for {i} {escapeCodes}");
+                        break;
+                    }
+
+                    Color color;
+                    bool isFore = escapeCodes[i] == 38;
+                    ++i;
+
+                    if (escapeCodes[i] == 5)
+                    {
+                        // ESC[38:5:⟨n⟩m Select foreground color
+                        ++i;
+                        bold = escapeCodes[i] is >= _boldOffset and < 2 * _boldOffset;
+                        color = Get8bitColor(escapeCodes[i], out currentColorId);
+                    }
+                    else if (escapeCodes[i] == 2)
+                    {
+                        // ESC[38;2;⟨r⟩;⟨g⟩;⟨b⟩ m Select RGB foreground color
+                        ++i;
+                        if (i >= escapeCodes.Count - 2)
+                        {
+                            Trace.WriteLine($"Unexpected too few arguments for {i} {escapeCodes}");
+                            DebugHelpers.Fail($"Unexpected too few arguments for {i} {escapeCodes}");
+                            break;
+                        }
+
+                        color = Color.FromArgb(escapeCodes[i], escapeCodes[++i], escapeCodes[++i]);
```

Comment:
(In C/C++, these `++i` would have unexpected results.)

Reply:
Change this to use + for all
The compiler should optimize in the same way.